Client/game_sa/CFxSA.cpp

```diff
@@ -239,3 +246,37 @@ void CFxSA::TriggerFootSplash(CVector& vecPosition)
         call    dwFunc
     }
 }
+
+bool CFxSA::IsShadowsLimitReached()
+{
+    unsigned short& CShadows_ShadowsStoredToBeRendered = *(unsigned short*)0xC403DC;
+    return CShadows_ShadowsStoredToBeRendered >= 48;
+}
+
+bool CFxSA::AddShadow(eShadowType shadowType, CVector& vecPosition, CVector2D& vecOffset1, CVector2D& vecOffset2, SColor color, float fZDistance,
```

Comment:
I think **vecPosition** must be const, as well as **vecOffset1** and **vecOffset2**